### PR TITLE
feat(chore): Document DesiredAccess construction

### PR DESCRIPTION
**Pull request by GHEMID-Mohamed**

### CODE REVIEW

File: lib/messages/create.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ module.exports = message({
       },
       request: {
         Buffer: buffer,
+        /**
+         * https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/77b36d0f-6016-458a-a7a0-0f4a72ae1534
+         * DesiredAccess is a construction of : SYNCHRONIZE, DELETE, READ_CONTROL, WRITE_DAC, FILE_WRITE_ATTRIBUTES, FILE_READ_ATTRIBUTES, FILE_DELETE_CHILD, FILE_WRITE_EA, FILE_READ_DATA, FILE_WRITE_DATA, FILE_APPEND_DATA, FILE_READ_EA
```
Comment (julien-f) [inline lib/messages/create.js]: Add this values in `constants` and combine them properly.

File: lib/messages/create.js [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+var WRITE_DAC = require('../structures/constants').WRITE_DAC;
+var FILE_WRITE_ATTRIBUTES = require('../structures/constants')
+  .FILE_WRITE_ATTRIBUTES;
+var FILE_READ_ATTRIBUTES = require('../structures/constants')
+  .FILE_READ_ATTRIBUTES;
+var FILE_DELETE_CHILD = require('../structures/constants').FILE_DELETE_CHILD;
+var FILE_WRITE_EA = require('../structures/constants').FILE_WRITE_EA;
+var FILE_READ_DATA = require('../structures/constants').FILE_READ_DATA;
+var FILE_APPEND_DATA = require('../structures/constants').FILE_APPEND_DATA;
+var FILE_READ_EA = require('../structures/constants').FILE_READ_EA;
```
Comment (julien-f) [inline lib/messages/create.js]: replace by `var constants = require('../structures/constants`)`

File: lib/messages/create.js [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+  DELETE |
+  READ_CONTROL |
+  WRITE_DAC |
+  FILE_WRITE_ATTRIBUTES |
+  FILE_READ_ATTRIBUTES |
+  FILE_DELETE_CHILD |
+  FILE_WRITE_EA |
+  FILE_READ_DATA |
+  FILE_APPEND_DATA |
+  FILE_READ_EA;
```
Comment (julien-f) [inline lib/messages/create.js]: Sort liens.

File: lib/messages/create.js [hunk 4]
```diff
@@ -22,7 +51,11 @@ module.exports = message({
       },
       request: {
         Buffer: buffer,
-        DesiredAccess: 0x001701df,
```
Comment (julien-f) [inline lib/messages/create.js:25]: What is the type of this value?

File: lib/messages/create.js [hunk 5]
```diff
@@ -22,7 +51,11 @@ module.exports = message({
...
+        /**
+         * https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/77b36d0f-6016-458a-a7a0-0f4a72ae1534
+         * DesiredAccess is a construction of : SYNCHRONIZE, DELETE, READ_CONTROL, WRITE_DAC, FILE_WRITE_ATTRIBUTES, FILE_READ_ATTRIBUTES, FILE_DELETE_CHILD, FILE_WRITE_EA, FILE_READ_DATA, FILE_WRITE_DATA, FILE_APPEND_DATA, FILE_READ_EA
+         */
+        DesiredAccess: hexTo0xFormat(desiredAccess.toString(16)),
```
Comment (julien-f) [inline lib/messages/create.js]: What is the type of this value?

File: lib/messages/create.js [hunk 6]
```diff
@@ -22,7 +51,11 @@ module.exports = message({
       },
       request: {
         Buffer: buffer,
-        DesiredAccess: 0x001701df,
+        /**
+         * https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/77b36d0f-6016-458a-a7a0-0f4a72ae1534
+         * DesiredAccess is a construction of : SYNCHRONIZE, DELETE, READ_CONTROL, WRITE_DAC, FILE_WRITE_ATTRIBUTES, FILE_READ_ATTRIBUTES, FILE_DELETE_CHILD, FILE_WRITE_EA, FILE_READ_DATA, FILE_WRITE_DATA, FILE_APPEND_DATA, FILE_READ_EA
+         */
```
Comment (julien-f) [inline lib/messages/create.js]: You can remove this comment, it's no longer necessary now that the value is explicit.

File: lib/structures/constants.js [hunk 7]
```diff
@@ -19,4 +19,15 @@ module.exports = {
...
+  DELETE: 0x00010000,
+  READ_CONTROL: 0x00020000,
+  WRITE_DAC: 0x00040000,
+  FILE_WRITE_ATTRIBUTES: 0x00000100,
+  FILE_READ_ATTRIBUTES: 0x00000080,
+  FILE_DELETE_CHILD: 0x00000040,
+  FILE_WRITE_EA: 0x00000010,
+  FILE_READ_DATA: 0x00000001,
+  FILE_APPEND_DATA: 0x00000004,
+  FILE_READ_EA: 0x00000008,
```
Comment (julien-f) [inline lib/structures/constants.js:26]: Sort lines.

Comment (julien-f) [reply lib/structures/constants.js:26]: Or at least, keep the order of the documentation.

File: lib/messages/create.js (hunk 4 above)
Comment (GHEMID-Mohamed) [reply lib/messages/create.js:25]: Number

**Review by julien-f — APPROVED** [review]

**Pull request merged**